### PR TITLE
For range fix and more tests

### PR DESCRIPTION
**Pull request by rphh**

### CODE REVIEW

File: tests/unit/translator/resources/for_with_range.pml [hunk 1]
```diff
@@ -24,6 +24,30 @@ inline Taskwithloop_0_transition(id)
...
+        x = 1;
+        do
+        ::(x < 3)->
+          MyInteger_assign_value(global_state.taskwithloop.y, (global_state.taskwithloop.y + x));
+          x = (x + 1);
+        ::else->
+          break;
+        od;
+      }
+      ;
```
Comment (Lurkerpas) [inline tests/unit/translator/resources/for_with_range.pml:38]: Can we think about eliminating this one?

File: tests/unit/translator/resources/for_with_range.pr [hunk 2]
```diff
@@ -48,6 +48,12 @@ process Taskwithloop;
             task for x in RANGE(0, 10, 2):
 y := y + x
 y := Y * 2
+endfor;
+            task for x in RANGE(1, 3):
+y := y + x
+endfor;
+            task for x in RANGE(4):
```
Comment (Lurkerpas) [inline tests/unit/translator/resources/for_with_range.pr:55]: Can we test in RANGE(length(somearray))?

File: tests/unit/translator/resources/for_with_range.pml (hunk 1 above)
Comment (rphh) [reply tests/unit/translator/resources/for_with_range.pml:38]: This looks like separate task

File: tests/unit/translator/resources/for_with_range.pr (hunk 2 above)
Comment (rphh) [reply tests/unit/translator/resources/for_with_range.pr:57]: added

**Pull request merged**